llvm/test/DebugInfo/CAS/AArch64/test_relocation_zero.ll

```diff
@@ -0,0 +1,64 @@
+;RUN: llc -O0 --filetype=obj --cas-backend  --cas=%t/cas  --mccas-casid %s -o %t/test.id
+;RUN: llvm-cas-dump --cas=%t/cas --casid-file %t/test.id --hex-dump | FileCheck %s
+
+;CHECK: mc:debug_info_cu llvmcas://{{[0-9a-z]+}}
+;CHECK-NEXT: 0x3c 0x00 0x00 0x00 0x04 0x00 0x00 0x00
+;CHECK-NEXT: 0x00 0x00 0x08 0x01 0x00 0x00 0x00 0x00
+;CHECK-NEXT: 0x0c 0x00 0x6a 0x00 0x00 0x00 0x00 0x00
+;CHECK-NEXT: 0x00 0x00 0x93 0x00 0x00 0x00 0x00 0x00
+;CHECK-NEXT: 0x00 0x00 0x00 0x00 0x00 0x00 0x04 0x00
+;CHECK-NEXT: 0x00 0x00 0x02 0x00 0x00 0x00 0x00 0x00
+;CHECK-NEXT: 0x00 0x00 0x00 0x04 0x00 0x00 0x00 0x01
+;CHECK-NEXT: 0x6f 0xb8 0x00 0x00 0x00 0x01 0x01 0x00
+;CHECK-NEXT: mc:debug_info_cu llvmcas://{{[0-9a-z]+}} 
+;CHECK-NEXT: 0x56 0x00 0x00 0x00 0x04 0x00 0x00 0x00
+;CHECK-NEXT: 0x00 0x00 0x08 0x01 0x00 0x00 0x00 0x00
+;CHECK-NEXT: 0x0c 0x00 0x6a 0x00 0x00 0x00 0x3c 0x00
+;CHECK-NEXT: 0x00 0x00 0x93 0x00 0x00 0x00 0x00 0x00
+;CHECK-NEXT: 0x00 0x00 0x00 0x00 0x00 0x00 0x18 0x00
+;CHECK-NEXT: 0x00 0x00 0x02 0x00 0x00 0x00 0x00 0x00
+;CHECK-NEXT: 0x00 0x00 0x00 0x18 0x00 0x00 0x00 0x01
+;CHECK-NEXT: 0x6f 0xbc 0x00 0x00 0x00 0x01 0x02 0x52
+;CHECK-NEXT: 0x00 0x00 0x00 0x03 0x02 0x91 0x0c 0xc4
+;CHECK-NEXT: 0x00 0x00 0x00 0x01 0x02 0x52 0x00 0x00
+;CHECK-NEXT: 0x00 0x00 0x04 0xc0 0x00 0x00 0x00 0x05
+;CHECK-NEXT: 0x04 0x00
+
+define void @foo() #0 !dbg !9 {
+  ret void, !dbg !14
+}
+define i32 @bar(i32 noundef %x) #0 !dbg !15 {
+entry:
+  %x.addr = alloca i32, align 4
+  store i32 %x, ptr %x.addr, align 4
+  call void @llvm.dbg.declare(metadata ptr %x.addr, metadata !20, metadata !DIExpression()), !dbg !21
+  %0 = load i32, ptr %x.addr, align 4, !dbg !22
+  %add = add nsw i32 %0, 1, !dbg !23
+  ret i32 %add, !dbg !24
+}
```

Comment:
Do we need this many instructions and two functions? I think that once I see the answer on the other comment I may understand the test a bit better